packages/firebase/index.d.ts

```diff
@@ -8289,12 +8289,109 @@ declare namespace firebase.firestore {
      */
     terminate(): Promise<void>;
 
+    /**
+     * Loads a Firestore bundle into the local cache.
+     *
+     * @param bundleData
+     *   An object representing the bundle to be loaded. Valid objects are `ArrayBuffer`,
+     *   `ReadableStream<Uint8Array>` or `string`.
+     *
+     * @return
+     *   A `LoadBundleTask` object, which notifies callers with progress updates, and completion
+     *   or error events. It can be used as a `Promise<LoadBundleTaskProgress>`.
+     */
+    loadBundle(
+      bundleData: ArrayBuffer | ReadableStream<Uint8Array> | string
+    ): LoadBundleTask;
+
+    /**
+     * Reads a Firestore `Query` from local cache, identified by the given name.
+     *
+     * The named queries are packaged  into bundles on the server side (along
+     * with resulting documents), and loaded to local cache using `loadBundle`. Once in local
+     * cache, use this method to extract a `Query` by name.
+     */
+    namedQuery(name: string): Promise<Query<DocumentData> | null>;
+
     /**
      * @hidden
      */
     INTERNAL: { delete: () => Promise<void> };
   }
 
+  /**
+   * Represents the task of loading a Firestore bundle. It provides progress of bundle
+   * loading, as well as task completion and error events.
+   *
+   * The API is compatible with `Promise<LoadBundleTaskProgress>`.
+   */
+  export interface LoadBundleTask extends PromiseLike<LoadBundleTaskProgress> {
```

Comment:
Need to create entries in https://github.com/firebase/firebase-js-sdk/blob/master/scripts/docgen/content-sources/js/toc.yaml for 2 new interfaces, LoadBundleTask and LoadBundleTaskProgress

Can just insert this alphabetically in the Firestore section:
```
  - title: "LoadBundleTask"
    path: /docs/reference/js/firebase.firestore.LoadBundleTask
  - title: "LoadBundleTaskProgress"
    path: /docs/reference/js/firebase.firestore.LoadBundleTaskProgress
```

If this is available for Node, also here:
https://github.com/firebase/firebase-js-sdk/blob/master/scripts/docgen/content-sources/node/toc.yaml

Reply:
Done.